### PR TITLE
Render index with Slim.

### PR DESCRIPTION
**Pull request by japfohl**
- Adds sufficient session checking for admin sessions.
- Switches admin site to admin site 1 rather than 2.
- Misc. bug fixes and changes to admin panel.
- add session checking to all appropriate pages
- move all testing stuff to root/tests directory
- move/rename test helper to ApiTestCase in src directory
- make sure all tests inherit from base test case
- create example template render and fix search query
- donors render in php
- fix header function to act as proper form
- begin process of moving all site routes into Website.php
- handle drawing map pins using data attribute
- update template file so there is no way phpunit will try and run as test

Notes:  

- You'll have to do a `php composer.phar update`.  
- The new index implementation breaks the old one and the links on that page but I can't figure out a better way to do it then to progressively break the old site and move functionality to the newer / better version until its all completely implemented.

To Test:

1. Checkout my branch.
2. update composer
3. Run the dev server
4. View the index page
5. Run the tests.
6. Critique the code! (Notice... I added newlines!)

### CODE REVIEW

File: public_html/js/home.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    });
+
+    // create the new map
+    var indexMap = corvallisMap();
+
+    // add data to the map
+    for (i = 0; i < mapJson.length; i++) {
+        // set the pin color based on the type of location
+        var pinColor;
+        if (mapJson[i].type === "recycle") {
```
Comment (istanfield09) [inline public_html/js/home.js]: kind of a barely there critique but you could create a map at the top of the file that defines the color per each type of location and just call `pinColor = colorMap[mapJson[i].type];`. That way the color definitions are close together, and more easily modified, extendable, etc..

File: src/lib/Util.php [hunk 2]
```diff
@@ -5,12 +5,36 @@
...
+    }
+
+    //replacing a single underscore with a slash
+    public static function underscoreToSlash(&$string) {
+
+        $string = str_replace("_","/", $string);
+    }
+
+    // Helper used to safely fetch values from associative arrays
+    public static function fetch_val($key, $array, $default = 'undefined') {
```
Comment (istanfield09) [inline src/lib/Util.php]: Nice

File: public_html/js/home.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+function initIndexMap() {
+    // get the node and the json data attribute
+    var map = document.querySelector('#map');
+    var mapJson = JSON.parse(map.dataset.maplocs);
+
+    // remove duplicates if there are any
+    mapJson = mapJson.sort(function(a, b) { return a.id - b.id }).filter(function(item, pos, arr) {
```
Comment (istanfield09) [inline public_html/js/home.js]: https://gph.is/1NOMJAr

File: templates/home.php [hunk 4]
```diff
@@ -1,6 +1,60 @@
...
+        "add" =>    Util::fetch_val('address_line_1', $r),
+        "city" =>   Util::fetch_val('city', $r),
+        "state"  => Util::fetch_val('abbreviation', $r),
+        "zip" =>    Util::fetch_val('zip_code', $r),
+        "id" =>     Util::fetch_val('id', $r),
+        "type" =>   "repair"
+    ];
+}
+
+// append reuse locaitons
```
Comment (istanfield09) [inline templates/home.php]: typo, nbd

File: public_html/js/home.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    var pincolors = {
+        "recycle": "7C903A",
+        "repair": "47A6B2",
+        "reuse": "F89420"
+    };
+
+    // add data to the map
+    for (i = 0; i < mapJson.length; i++) {
+        // create the pins and data for the marker
+        var pinImage = pin(pincolors[mapJson[i].type]);
```
Comment (istanfield09) [inline public_html/js/home.js]: clean af

Comment (japfohl) [reply public_html/js/home.js]: agreed!

**Pull request merged**